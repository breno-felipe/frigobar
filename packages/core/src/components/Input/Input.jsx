import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const InputContainer = styled.div`
  position: relative;
  flex: 1 1 0%;

  ${({ icon }) =>
    icon
      ? `
    position: absolute;
    left: 16px;
;
    top: 50%;
    transform: translateY(-50%);
    fill: rgb(40, 39, 44);
    font-size: 14px;
    transition: fill 0.2s ease 0s;
  `
      : ``}
`;

const InputComponent = styled.input`
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0px 1em 0px 2.65em;
  border-radius: 5px;

  ${({
    theme: {
      components: {
        input: { backgroundColor, sizes, textColor },
      },
    },
    skin,
    full,
    disabled,
    size,
  }) => `
    background-color: ${backgroundColor[skin]};
    color: ${textColor.enabled};
    font-size: ${sizes[size].font}rm;
    padding: ${sizes[size].padding.top}px ${sizes[size].padding.right}px ${
    sizes[size].padding.bottom
  }px ${sizes[size].padding.left}px;
    ${
      disabled
        ? `
      background-color: ${backgroundColor.disabled};
      color: ${textColor.disabled};
      cursor: not-allowed;
    `
        : `
        &:hover {
          box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.4);
        }
        `
    }

    ${full ? 'width: 100%;' : ''}
  `}
`;

const Input = forwardRef(
  ({ type, full, theme, size, disabled, icon, ...props }, ref) => (
    <InputContainer>
      {icon}
      <InputComponent
        theme={theme}
        full={full}
        size={size}
        disabled={disabled}
        type={type || 'text'}
        ref={ref}
        {...props}
      />
    </InputContainer>
  ),
);

Input.propTypes = {
  type: PropTypes.string,
  icon: PropTypes.string,
  full: PropTypes.bool,
  disabled: PropTypes.bool,
  skin: PropTypes.oneOf(['primary', 'neutral']),
  size: PropTypes.oneOf(['large', 'medium', 'small']),
};

Input.defaultProps = {
  type: 'text',
  full: false,
  disabled: false,
  icon: undefined,
  skin: 'primary',
  size: 'small',
};

export default Input;
