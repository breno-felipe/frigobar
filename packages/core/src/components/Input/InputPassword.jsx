import React, { forwardRef, useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Visibility, VisibilityOff } from '@material-ui/icons';

const InputContainer = styled.div`
  position: relative;
  flex: 1 1 0%;

  svg {
    position: absolute;
    left: 14px;
    top: 50%;
    transform: translateY(-50%);
    fill: rgb(40, 39, 44);
    font-size: 1.3rem;
    transition: fill 0.2s ease 0s;
  }
`;

const InputComponent = styled.input`
  height: 35px;
  font-size: 14px;
  padding: 0px 1em 0px 3.35em;
  border-radius: 5px;
  border: 1px solid #000;

  ${({
    theme: {
      components: {
        input: { backgroundColor, textColor },
      },
    },
    skin,
    full,
    disabled,
  }) => `
    background-color: ${backgroundColor[skin]};
    color: ${textColor.enabled};
    ${
      disabled
        ? `
      background-color: ${backgroundColor.disabled};
      color: ${textColor.disabled};
      cursor: not-allowed;
    `
        : `
        &:hover {
          box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.4);
        }
        `
    }

    ${full ? 'width: 100%;' : ''}
  `};
`;

const InputPassword = forwardRef(
  ({ type, full, theme, disabled, icon, ...props }, ref) => {
    const [showPassword, toggleShowPassword] = useState(false);
    const inputRef = useRef(null);

    useEffect(() => {
      const { current: element } = inputRef;

      if (element.value.length) {
        element.setSelectionRange(element.value.length, element.value.length);
      }
    }, [showPassword]);

    const togglePassword = e => {
      if (e.type === 'click') {
        inputRef.current.focus();
      }

      e.preventDefault();
      toggleShowPassword(!showPassword);
    };

    return (
      <InputContainer>
        <InputComponent
          theme={theme}
          full={full}
          disabled={disabled}
          type={showPassword ? 'text' : 'password'}
          ref={inputRef || ref}
          icon={icon}
          {...props}
        />

        {showPassword ? (
          <Visibility
            width={20}
            height={20}
            onClick={togglePassword}
            role="button"
          />
        ) : (
          <VisibilityOff
            width={20}
            height={20}
            onClick={togglePassword}
            role="button"
          />
        )}
      </InputContainer>
    );
  },
);

InputPassword.propTypes = {
  type: PropTypes.string,
  icon: PropTypes.string,
  full: PropTypes.bool,
  disabled: PropTypes.bool,
  skin: PropTypes.oneOf(['primary', 'neutral']),
};

InputPassword.defaultProps = {
  type: 'text',
  full: false,
  disabled: false,
  icon: undefined,
  skin: 'primary',
};

export default InputPassword;
