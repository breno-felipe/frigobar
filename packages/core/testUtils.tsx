import React from 'react';
import PropTypes from 'prop-types';
import {
  render,
  fireEvent,
  waitFor,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from './src/theme';

const Providers = ({ children }: { children?: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

Providers.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, fireEvent, waitFor };
