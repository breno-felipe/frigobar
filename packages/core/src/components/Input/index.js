import Input from './Input';
import InputPassword from './InputPassword';

export default { InputPassword, Input };
