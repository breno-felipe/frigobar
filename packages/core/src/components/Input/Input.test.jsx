import React from 'react';
import { render } from '../../../testUtils';

import Input from './Input';

describe('<Input />', () => {
  describe('Snapshpts', () => {
    it('should match snapshot with all skins', () => {
      const inputs = [
        <Input skin="primary" />,
        <Input skin="primary" />,
        <Input skin="primary" />,
      ];

      inputs.forEach(Component => {
        const { container } = render(Component);
        expect(container).toMatchSnapshot();
      });
    });
    it('should match snapshot with input disabled', () => {
      const { container } = render(<Input disabled />);
      expect(container).toMatchSnapshot();
    });
    it('should match snapshot with all sizes', () => {
      const inputs = [
        <Input size="small" />,
        <Input size="medium" />,
        <Input size="large" />,
      ];

      inputs.forEach(Component => {
        const { container } = render(Component);
        expect(container).toMatchSnapshot();
      });
    });
  });
});
