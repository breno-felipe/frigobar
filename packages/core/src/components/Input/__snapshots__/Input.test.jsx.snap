// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshpts should match snapshot with all sizes 1`] = `
.c0 {
  position: relative;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
}

.c1 {
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0px 1em 0px 2.65em;
  border-radius: 5px;
  background-color: #598aed;
  color: #fdfdfd;
  font-size: 0.875rm;
  padding: 4px 12px 4px 12px;
}

.c1:hover {
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.4);
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      type="text"
    />
  </div>
</div>
`;

exports[`<Input /> Snapshpts should match snapshot with all sizes 2`] = `
.c0 {
  position: relative;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
}

.c1 {
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0px 1em 0px 2.65em;
  border-radius: 5px;
  background-color: #598aed;
  color: #fdfdfd;
  font-size: 1rm;
  padding: 8px 16px 8px 16px;
}

.c1:hover {
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.4);
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      type="text"
    />
  </div>
</div>
`;

exports[`<Input /> Snapshpts should match snapshot with all sizes 3`] = `
.c0 {
  position: relative;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
}

.c1 {
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0px 1em 0px 2.65em;
  border-radius: 5px;
  background-color: #598aed;
  color: #fdfdfd;
  font-size: 1.275rm;
  padding: 12px 20px 12px 20px;
}

.c1:hover {
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.4);
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      type="text"
    />
  </div>
</div>
`;

exports[`<Input /> Snapshpts should match snapshot with all skins 1`] = `
.c0 {
  position: relative;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
}

.c1 {
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0px 1em 0px 2.65em;
  border-radius: 5px;
  background-color: #598aed;
  color: #fdfdfd;
  font-size: 0.875rm;
  padding: 4px 12px 4px 12px;
}

.c1:hover {
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.4);
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      type="text"
    />
  </div>
</div>
`;

exports[`<Input /> Snapshpts should match snapshot with all skins 2`] = `
.c0 {
  position: relative;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
}

.c1 {
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0px 1em 0px 2.65em;
  border-radius: 5px;
  background-color: #598aed;
  color: #fdfdfd;
  font-size: 0.875rm;
  padding: 4px 12px 4px 12px;
}

.c1:hover {
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.4);
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      type="text"
    />
  </div>
</div>
`;

exports[`<Input /> Snapshpts should match snapshot with all skins 3`] = `
.c0 {
  position: relative;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
}

.c1 {
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0px 1em 0px 2.65em;
  border-radius: 5px;
  background-color: #598aed;
  color: #fdfdfd;
  font-size: 0.875rm;
  padding: 4px 12px 4px 12px;
}

.c1:hover {
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.4);
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      type="text"
    />
  </div>
</div>
`;

exports[`<Input /> Snapshpts should match snapshot with input disabled 1`] = `
.c0 {
  position: relative;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
}

.c1 {
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0px 1em 0px 2.65em;
  border-radius: 5px;
  background-color: #598aed;
  color: #fdfdfd;
  font-size: 0.875rm;
  padding: 4px 12px 4px 12px;
  background-color: undefined;
  color: #a8a8a8;
  cursor: not-allowed;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      disabled=""
      type="text"
    />
  </div>
</div>
`;
